scripts/generateOverview.java

```diff
@@ -0,0 +1,130 @@
+	Path templateFile = Path.of("wiki/SimRel/template.md");
+	Path overviewFile = templateFile.resolveSibling(simRelName + ".md");
+
+	SimRelDates dates = parseDatesJSON(Path.of("wiki/SimRel/" + simRelName + "_dates.json"));
+	performSanityChecks(simRelName, dates);
+
+	String simRelMonthYear = DateTimeFormatter.ofPattern("MMMM yyyy", LOCAL).format(dates.GA);
+	String previousSimRelName = DateTimeFormatter.ofPattern("yyyy-MM", LOCAL).format(dates.GA.minusMonths(3));
+	if (!Files.isRegularFile(overviewFile.resolveSibling(previousSimRelName + ".md"))) {
+		throw new IllegalStateException("Previous SimRel " + previousSimRelName + " not found.");
+	}
+	var calendarDate = DateTimeFormatter.ofPattern("yyyyMMdd", LOCAL);
+	String calendarPeriodStart = calendarDate.format(dates.M1.minusWeeks(2));
+	String calendarPeriodEnd = calendarDate.format(dates.GA.plusDays(1));
+
+	String content = Files.readString(templateFile);
+	content = content.replace("${SIMREL_NAME}", simRelName) //
+			.replace("${PREVIOUS_SIMREL_NAME}", previousSimRelName) //
+			.replace("${SIMREL_FULL_MONTH_YEAR}", simRelMonthYear)
+			.replace("${QUIET_PERIOD_START_SHORT}", SHORT.format(dates.RC2.plusDays(1)))
+			.replace("${QUIET_PERIOD_END_SHORT}", SHORT.format(dates.GA.minusDays(1)))
+			.replace("${CALENDAR_PERIOD}", calendarPeriodStart + "%2F" + calendarPeriodEnd);
+	content = resolveDateVariables(content, "M1", dates.M1);
+	content = resolveDateVariables(content, "M2", dates.M2);
+	content = resolveDateVariables(content, "M3", dates.M3);
+	content = resolveDateVariables(content, "RC1", dates.RC1);
+	content = resolveDateVariables(content, "RC2", dates.RC2);
+	content = resolveDateVariables(content, "GA", dates.GA);
+
+	Files.writeString(overviewFile, content);
+
+	IO.println("Overview for " + simRelName + " generated successfully: " + overviewFile);
+}
+
+String readSimRelName(String[] args) {
+	if (args.length < 1) {
+		throw new IllegalArgumentException("Expected SimRel name argument is missing");
+	}
+	String simRelName = args[0];
+	if (!Pattern.compile("(?<year>\\d{4})-(?<month>\\d{2})").matcher(simRelName).matches()) {
+		throw new IllegalArgumentException("SimRel name has unexpected format: " + simRelName);
+	}
+	return simRelName;
+}
+
+SimRelDates parseDatesJSON(Path file) throws IOException {
+	// Use built-in JSON parser once the JDK provides one
+	String content = Files.readString(file).trim();
+	if (!content.startsWith("{") || !content.endsWith("}")) {
+		throw new IllegalArgumentException("Unexpected content of" + file.getFileName());
+	}
+	var jsonDateObject = Pattern.compile("\"(?<name>\\w+)\"\\s*:\\s*\"(?<date>[^\"]+)\"\\s*(,|})");
+	var dates = jsonDateObject.matcher(content).results()
+			.collect(Collectors.toMap(m -> m.group("name"), m -> LocalDate.parse(m.group("date"))));
+	return new SimRelDates(getDate(dates, "M1"), getDate(dates, "M2"), getDate(dates, "M3"), //
+			getDate(dates, "RC1"), getDate(dates, "RC2"), getDate(dates, "GA"));
+}
```

Comment:
I also consider to use a properties file. And while it would be simpler to process in java other languages like Java-script don't seem to have as straight forward facilities as Java does.
And as there is the intention to provide a JSON parser and writer to the Java standard library in the future, this might be the more modern solution.
On the other hand a simple properties file could even be read using the 'source' command in a bash script.

For the use-case I have to for the machine readable dates file, both works fine as Jenkins provides simple steps for both, to read properties and JSON files.